### ISSUE TITLE
[MLS Spec change] Clarify tree hash and parent hash (#715)

### ISSUE THREAD

**Issue by raphaelrobert** [openmls/openmls#948]
**Link to the exact changes**
https://github.com/mlswg/mls-protocol/pull/715

**Description of the changes**
> Closes #655 
 > 
 > Depends on #694 
 > Conflicts with #713 
 > 
 > The diff here is messy because it includes #694.  The biggest change is the extended example, for which you can just scroll to the bottom.

**Comment by franziskuskiefer** [comment]
Tracked in #995

**Comment by raphaelrobert** [comment]
This is superseded by #995.
